frappe/core/doctype/doctype/doctype.py

```diff
@@ -290,9 +290,16 @@ def on_update(self):
 
 		self.update_fields_to_fetch()
 
-		from frappe import conf
-		allow_doctype_export = frappe.flags.allow_doctype_export or (not frappe.flags.in_test and conf.get('developer_mode'))
-		if not self.custom and not frappe.flags.in_import and allow_doctype_export:
+		allow_doctype_export = (
+			not self.custom
+			and not frappe.flags.in_import
+			and (
+				frappe.conf.developer_mode
+				or frappe.flags.allow_doctype_export
+				or frappe.flags.in_test
```

Comment:
I think this refactor changes the meaning a bit. Our priority was to allow export if it is exportable (`frappe.flags.allow_doctype_export`). If not exportable allow if it is in developer mode , but not if it is `in_test`.

The change makes it so that the three factors are either or. Unsure about this